src/symbolize/symbolizer.rs

```diff
@@ -194,6 +219,9 @@ pub struct Builder {
     /// mapping to members of an APK.
     #[cfg(feature = "apk")]
     apk_dispatch: Option<Dbg<Box<dyn ApkDispatch>>>,
+    /// The "dispatch" function to use when symbolizing addresses
+    /// mapping to members of a process.
+    process_dispatch: Option<Dbg<Box<dyn ProcessDispatch>>>,
```

Comment:
I think Rust supports default implementation for trait methods, right?

Asking because an alternative here would be to define single "Dispatcher" trait with two methods: one for apk and one for process, and user can choose to implement (or return non-None result) for either or both cases. Then on the building side there will be just one "set_dispatcher()" call.

Are there any downsides to keeping all possible kinds of dispatcher factories under single trait?

Reply:
In my mental model the `ProcessDispatch` trait isn't really a trait in the stricter sense, where the user implements something. It's just a different name for a function with the signature
```rust
Fn(ProcessMemberInfo<'_>) -> Result<Option<Box<dyn Resolve>>>
```
see:
```rust
pub trait ProcessDispatch: Fn(ProcessMemberInfo<'_>) -> Result<Option<Box<dyn Resolve>>> {}
```
And because it is auto implemented for each function with exactly that signature, the two are effectively equivalent:
```rust
impl<F> ProcessDispatch for F where F: Fn(ProcessMemberInfo<'_>) -> Result<Option<Box<dyn Resolve>>> {}
```

So conceptually, think of it more just as a (potentially stateful) function, nothing else. We have a trait only because trait aliases aren't stable yet and I *really* wanted to avoid writing the mouthful `Fn(ApkMemberInfo<'_>) -> Result<Option<Box<dyn Resolve>>>` everywhere. I have a TODO for `ApkDispatch`:
```rust
// TODO: Use a trait alias once stable.
#[cfg(feature = "apk")]
pub trait ApkDispatch: Fn(ApkMemberInfo<'_>) -> Result<Option<Box<dyn Resolve>>> {}
```

Once said trait aliases are stable, this would just be:
```rust
trait ApkDispatch = Fn(ApkMemberInfo<'_>) -> Result<Option<Box<dyn Resolve>>>;
```
(or whatever the exact syntax) and then it would truly just be a different (shorter) name.

Personally I'd want to keep it that way. In my opinion a user just providing a function is easy to understand and the most flexible for them (can be defined as a true `fn` or a closure, assigned to a variable in advance or be passed to the `set_*dispatch` call inline, etc.).

Reply:
I see, fair enough, mostly brought this up to understand tradeoffs myself, thanks.